<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1094</width>
    <height>917</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" rowstretch="3,1">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="1,3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Управление</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0,0">
           <property name="spacing">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <widget class="QLineEdit" name="lineEdit">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>Целевое значение</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>Отправить</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_3">
             <property name="text">
              <string>Старт/Стоп</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>15</height>
              </size>
             </property>
             <property name="text">
              <string>Кол-во отсчетов(50-1000)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox">
             <property name="minimum">
              <number>50</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>Циклическое значение</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Тип сигнала</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxCyclicSignalType">
             <item>
              <property name="text">
               <string>Треугольная волна</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Синусоида</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Прямоугольная волна</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Случайные значения</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Минимум</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="doubleSpinBoxCyclicMin">
             <property name="minimum">
              <double>-100.000000000000000</double>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Максимум</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="doubleSpinBoxCyclicMax">
             <property name="minimum">
              <double>-100.000000000000000</double>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>10.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Шаг</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="doubleSpinBoxCyclicStep">
             <property name="minimum">
              <double>0.100000000000000</double>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Период (мс)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBoxCyclicPeriod">
             <property name="minimum">
              <number>100</number>
             </property>
             <property name="maximum">
              <number>60000</number>
             </property>
             <property name="singleStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>1000</number>
             </property>
             <property name="displayIntegerBase">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxCyclicTargetEnable">
             <property name="text">
              <string>Включить циклическое задание</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Сеть</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,0,0">
           <property name="spacing">
            <number>1</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <item>
            <widget class="QLineEdit" name="lineEdit_2">
             <property name="placeholderText">
              <string>IP приема</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>15</height>
              </size>
             </property>
             <property name="text">
              <string>Порт приема</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_2">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100000</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_3">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>IP отправки</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>15</height>
              </size>
             </property>
             <property name="text">
              <string>Порт отправки</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_3">
             <property name="maximum">
              <number>100000</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3"/>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,3" columnstretch="1,1">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
      </property>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="title">
         <string>Экспоненциальный</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="doubleSpinBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
           <property name="value">
            <double>0.300000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="checkBoxExponentialEnable">
           <property name="text">
            <string>ON/OFF</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="labelExponentialAlpha">
           <property name="text">
            <string>Размер окна</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelExponentialStats">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelExponentialMemory">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="title">
         <string>Скользящее среднее</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinBoxMovingAverageWindow">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="labelMovingAverageWindow">
           <property name="text">
            <string>Размер окна</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelMovingAverageStats">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="checkBoxMovingAverageEnable">
           <property name="text">
            <string>ON/OFF</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelMovingAverageMemory">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="title">
         <string>Медианный</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="1">
          <widget class="QLabel" name="labelMedianWindow">
           <property name="text">
            <string>Размер окна</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="checkBoxMedianEnable">
           <property name="text">
            <string>ON/OFF</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinBoxMedianWindow">
           <property name="minimum">
            <number>3</number>
           </property>
           <property name="maximum">
            <number>51</number>
           </property>
           <property name="singleStep">
            <number>2</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelMedianStats">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelMedianMemory">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QGroupBox" name="groupBox_6">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="title">
         <string>Калмана</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Q</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="checkBoxKalmanEnable">
           <property name="text">
            <string>ON/OFF</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>R</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>P</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QDoubleSpinBox" name="doubleSpinBoxKalmanQ"/>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="doubleSpinBoxKalmanP"/>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="doubleSpinBoxKalmanR"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelKalmanMemory">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
